packages/peregrine/lib/talons/RootComponents/Category/category.gql.js

```diff
@@ -0,0 +1,71 @@
+import { gql } from '@apollo/client';
+
+export const GET_CATEGORY = gql`
+    query category(
+        $id: Int!
+        $pageSize: Int!
+        $currentPage: Int!
+        $filters: ProductAttributeFilterInput!
+        $sort: ProductAttributeSortInput
+    ) {
+        category(id: $id) {
+            id
+            description
+            name
+            product_count
+            meta_title
+            meta_keywords
+            meta_description
+        }
+        products(
+            pageSize: $pageSize
+            currentPage: $currentPage
+            filter: $filters
+            sort: $sort
+        ) {
+            items {
+                # id is always required, even if the fragment includes it.
+                id
+                # TODO: Once this issue is resolved we can use a
+                # GalleryItemFragment here:
+                # https://github.com/magento/magento2/issues/28584
+                name
+                price {
+                    regularPrice {
+                        amount {
+                            currency
+                            value
+                        }
+                    }
+                }
+                small_image {
+                    url
+                }
+                url_key
+                url_suffix
+            }
+            page_info {
+                total_pages
+            }
+            total_count
+        }
+    }
+`;
+
+export const GET_FILTER_INPUTS = gql`
+    query GetFilterInputsForCategory {
```

Comment:
I attempted along the way to make sure query names were unique, as I thought there was a lint rule about this. I've found a few duplicate names that I missed, but the linter doesn't care. Is this something we need to be mindful of?

Reply:
Hmm - I think this might be related to caching, but also readability in network tab/query browsers. You might try naming two queries identically but having them request different data and see what happens in the cache. Would need to have identical inputs as well.